types/validator_set.go

```diff
@@ -368,63 +368,84 @@ func (vals *ValidatorSet) Iterate(fn func(index int, val *Validator) bool) {
 
 // Verify that +2/3 of the set had signed the given signBytes.
 func (vals *ValidatorSet) VerifyCommit(chainID string, blockID BlockID, height int64, commit *Commit) error {
+
```

Comment:
IMO it would be cleaner to separate out most of the logic here into a stateless function (or a set of smaller stateless functions) to check commit validity, which then could be exported & re-used downstream (e.g. in the SDK light client).